### PR TITLE
build: manually clone advisory-db repo

### PR DESCRIPTION
**Pull request by domodwyer**
Fix `cargo deny` in CI.

---

* build: manually clone advisory-db repo (39d383a90)

      Use the git CLI to pull the advisory-db repo, instead of letting cargo-deny do
      it as it seems to ignore the "pull with CLI" cargo setting:

          https://github.com/EmbarkStudios/cargo-deny/issues/388

      Potentially related to GitHub killing old key support today:

          https://github.blog/2021-09-01-improving-git-protocol-security-github/

### CODE REVIEW

**Review by crepererum — APPROVED** [review]
Looks similar to what we had for `cargo-audit`.

**Comment by domodwyer** [comment]
Yeah I guess they have their own libgit2 linked in or something that ignores the cargo config 🤷

File: .circleci/config.yml [hunk 1]
```diff
@@ -124,9 +124,17 @@ jobs:
...
           name: Install cargo-deny
           command: cargo install --force cargo-deny
+      - run:
+          # Manually pull the advisory DB with the git CLI as cargo deny seems
+          # to ignore CARGO_NET_GIT_FETCH_WITH_CLI.
+          #
+          # https://github.com/EmbarkStudios/cargo-deny/issues/388
+          name: Pull advisory-db
+          command: |
+            git clone --depth=1 https://github.com/RustSec/advisory-db.git ~/.cargo/advisory-dbs/github.com-dfdf83aac62a7e5e
```
Comment (crepererum) [inline .circleci/config.yml:134]: Is this the local path stable? I'm wondering if `cargo-deny` just does nothing (i.e. doesn't audit at all) if we clone into the wrong target path.

Comment (domodwyer) [reply .circleci/config.yml:134]: https://app.circleci.com/pipelines/github/influxdata/influxdb_iox/16834/workflows/d0cd42bc-30df-4f26-a7eb-329dfcf2e453/jobs/136009

**Pull request merged**